fuel-core/src/executor.rs

```diff
@@ -103,20 +105,23 @@ impl Executor {
 
             // update block commitment
             let tx_fee = self.total_fee_paid(tx, vm_result.receipts())?;
-            // TODO: use SMT instead of this manual approach
+            // TODO: use merkle-sum-tree instead of this manual hashing approach
```

Comment:
This should just be a plain binary Merkle tree. I will update the block header specs soon for this.

Reply:
Ah ok, well that should be an easy change.